README.md

```diff
@@ -50,6 +50,22 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. `->` (arrow) expresses a function's explicit parameter and return value types. `String` -> `[String]` -> `[String]` is a function which takes a `String` and an `[String]` and returns an `[String]`.
```

Comment:
Haskell's special `[]` type constructor is not something we should emulate, in my view. In Sanctuary we've switched to using `Array :: Type -> Type` (as PureScript did quite a while ago), as it's more consistent:

```purescript
maybeToArray :: Maybe a -> [a]
arrayToMaybe :: [a] -> Maybe a
```

```purescript
maybeToArray :: Maybe a -> Array a
arrayToMaybe :: Array a -> Maybe a
```

Before introducing `->` via `String -> Array String -> Array String` it would be good to introduce type constructors so readers know what `Array String` represents.